src/CLI/actioner/sonic-cli-neighbors.py

```diff
@@ -51,57 +54,38 @@ def get_keypath(func,args):
 
     return keypath, body
 
-def fdb_call(macAddr, vlanName):
-    aa = cc.ApiClient()
-
-    vlanId = vlanName[len("Vlan"):]
-    macAddr = macAddr.strip()
-
-    keypath = cc.Path('/restconf/data/openconfig-network-instance:network-instances/network-instance={name}/fdb/mac-table/entries/entry={macaddress},{vlan}', name='default', macaddress=macAddr, vlan=vlanId)
+def build_mac_list():
+    global macDict
+    keypath = cc.Path('/restconf/data/openconfig-network-instance:network-instances/network-instance={name}/fdb/mac-table/entries', name='default')
 
     try:
         response = aa.get(keypath)
         response = response.content
 
-        if 'openconfig-network-instance:entry' in response.keys():
-                instance = response['openconfig-network-instance:entry'][0]['interface']['interface-ref']['state']['interface']
-
-        if instance is not None:
-                return instance
-        return "-"
+        macContainer = response.get('openconfig-network-instance:entries')
+        macList = macContainer.get('entry')
 
-    except:
-        return "-"
-
-def process_single_nbr(response, args):
-    nbr_list = []
-    ext_intf_name = "-"
-    nbr = response['openconfig-if-ip:neighbor']
+        for macEntry in macList:
+            vlan = macEntry.get('vlan')
+            if vlan is None:
```

Comment:
Can you optimize this further? see whether looping all MAC entries and check VLAN and MAC can be avoided with just one lookup for both MAC and VLAN together since these two are keys in OC.

Reply:
That was the case earlier and much slower than this approach.  For reference see the 'get_egress_port' method of the old file.